src/domain/task-system/services/src/task_scheduler_impl.rs

```diff
@@ -52,11 +52,9 @@ impl TaskSchedulerImpl {
 impl TaskScheduler for TaskSchedulerImpl {
     #[tracing::instrument(level = "info", skip_all, fields(?logical_plan))]
     async fn create_task(&self, logical_plan: LogicalPlan) -> Result<TaskState, CreateTaskError> {
-        let mut task = Task::new(
-            self.time_source.now(),
-            self.event_store.new_task_id().await?,
-            logical_plan,
-        );
+        let new_task_id = self.event_store.new_task_id().await?;
+        let mut task = Task::new(self.time_source.now(), new_task_id, logical_plan);
```

Comment:
`TaskSchedulerImpl` is a singleton - it is created once and lives until program terminates. It's made a singleton to maintain an in-memory task queue.

Because it's a singleton - all references it holds, e.g. `event_store`, will live forever too, but this won't work well if store needs a DB transaction.

I think since Scheduler is a "top level" component it should manage transactions manually and create event store in transactional scope when it's needed.

Reply:
Scheduler is really a singleton, but it's basically an API to manage a queue, rather than an actively running component that should control transactions. Maybe it should receive the `event_store` reference as an argument - or transactional catalog.